velox/exec/PartitionedOutput.cpp

```diff
@@ -109,6 +109,21 @@ BlockingReason Destination::flush(
 }
 } // namespace detail
 
+namespace {
+// If the upstream is partial limit, downstream is final limit and we want to
+// flush as soon as we can to reach the limit and do as less work as possible.
+bool eagerFlush(const core::PlanNode& node) {
```

Comment:
I wonder if it makes sense to move this logic into LocalPlanner and have PartitionedOutput operator take eagerFlush as a constructor argument. 

Also, wondering if we should check the limit value and enable eager flush only if limit is relatively low, e.g. exclude queries like SELECT * FROM t LIMIT 10000000000.